_rules/table-header-cell-has-assigned-cells-d0f69e.md

```diff
@@ -24,34 +25,37 @@ acknowledgments:
 
 ## Applicability
 
-This rule applies to any [HTML element][] with a [semantic][semantic role] [rowheader][] or [columnheader][] for which all of the following is true:
+This rule applies to any non-empty `th` element that is [included in the accessibility tree][], has a [semantic role][] of [rowheader][] or [columnheader][], and for which all of the following are true:
 
-- the element is [visible][]; and
-- the element is [included in the accessibility tree][]; and
-- the element has at least one ancestor in the [flat tree][] that is a [semantic][semantic role] [table][] or [grid][]; and
-- the element's closest ancestor in the [flat tree][] that is a [semantic][semantic role] [table][] or [grid][] is [included in the accessibility tree][].
+- **in a table** the `th` element is a [descendant][] of a `table` element for which all of the following are true:
+  - **visible** the `table` element is [visible][]; and
+  - **minimum rows** the `table` element's [`rows` collection](https://html.spec.whatwg.org/multipage/tables.html#dom-table-rows) has 2 or more `tr` elements; and
+  - **non-empty data cell** the `table` element has at least one non-empty `td` element that is a [child][] of a `tr` element with the [`rowIndex`][rowindex] value of 1 or more if the `tr` element with the value of 0 has at least two `th` [children][child] or the value of 0 or more if the `tr` element with the value of 0 has one or less `th` [children][child].
+- **no explicit role** none of the elements that make up the [table model][] have an [explicit semantic role][].
```

Comment:
What if the explicit role is the same as the implicit one? (`<th role="rowheader">`)
Plus, this seems to be a bullet in the outer list (used by the `th`) but rather refers to the `table` (i.e. the inner list).

Reply:
I've chosen the safer version as we don't know for sure what browsers do when aria roles are added as it's not in the spec. UA have their own way of assigning headers when aria is used and at the moment we can only guess. I understand that the same implicit role would work fine in practice but I'm slightly concerned we may expose the rule to some weird, uncommon example that will be a false positive.

Do you think the rule will benefit from adding an exception for the same explicit role?

Hmm... this role applies to all the elements that make up the test target's table element, so `table`,`thead`, `tbody`,`tr`, etc.  I thought that the [table model](https://html.spec.whatwg.org/multipage/tables.html#table-model) definition was explicit enough. Open to suggestions on this.